packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts

```diff
@@ -40,6 +40,81 @@ describe('testing search module actions', () => {
     jest.clearAllMocks();
   });
 
+  describe('batchStateResetsAfterRequestUpdate', () => {
+    // eslint-disable-next-line max-len
+    it('should not reset the page when the page parameter of the request changes', async () => {
+      resetSearchStateWith(store, { query: 'lego', page: 2 });
+      await store.dispatch('batchStateResetsAfterRequestUpdate', {
+        newRequest: {
+          query: 'lego',
+          page: 3
+        },
+        oldRequest: store.getters.request!
+      });
+
+      expect(store.state).toHaveProperty('page', 2);
```

Comment:
Even if it is more verbose, can we check the full final request object in these tests? It would be much safer than just checking one or two properties.

Reply:
Sure. Just take into account that the `newRequest` parameter is not going to be the same as the expected object, as this action just resets some parameters based on the different keys between the new and old request.